tests/Unit/PointwiseFunctions/GeneralRelativity/Test_KerrSchildCoords.cpp

```diff
@@ -0,0 +1,52 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <algorithm>
+#include <limits>
+#include <tuple>
+
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/Tensor/Tensor.hpp"  // IWYU pragma: keep
+#include "PointwiseFunctions/GeneralRelativity/KerrSchildCoords.hpp"
+#include "tests/Unit/Pypp/CheckWithRandomValues.hpp"
+#include "tests/Unit/Pypp/SetupLocalPythonEnvironment.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+
+namespace {
+
+template <typename DataType>
+void test_kerr_schild_coords(const DataType& used_for_size) noexcept {
+  const double spin_parameter = 0.9375;
+  const auto member_variables = std::make_tuple(spin_parameter);
+  gr::KerrSchildCoords kerr_schild_coords(spin_parameter);
+
+  pypp::check_with_random_values<1>(
+      &gr::KerrSchildCoords::r_coord_squared<DataType>, kerr_schild_coords,
+      "TestFunctions", "r_coord_squared", {{{-10.0, 10.0}}}, member_variables,
+      used_for_size);
+
+  pypp::check_with_random_values<1>(
+      &gr::KerrSchildCoords::cartesian_from_spherical<DataType>,
+      kerr_schild_coords, "TestFunctions", "cartesian_from_spherical",
+      {{{-10.0, 10.0}}}, member_variables, used_for_size);
+}
```

Comment:
Is it possible (and reasonable) to also test a known analytic value, like a specific point in the spacetime?

Reply:
Probably, but we need this merged 2 weeks ago, so there should be an issue created and done in a separate PR :)

Reply:
Given the number of bugs in this PR, I'd be more comfortable with the analytic solution being implemented now. It should test all 3 coord axes (both +th and -th sides), and any points where trig functions used become zero or infinite. I think that should provide a thorough test suite, like @wthrowe requested.